src/Jaeger.Core/Reporters/RemoteReporter.cs

```diff
@@ -88,25 +108,34 @@ public async Task CloseAsync(CancellationToken cancellationToken)
             }
         }
 
+        internal void FlushOnTimer(object sender, ElapsedEventArgs e)
+        {
+            try
+            {
+                Flush();
+            }
+            finally
+            {
+                _timer.Start();
+            }
+        }
+
+        internal async Task FlushAsync(CancellationToken token)
+        {
+            await Task.Run(Flush, token);
+        }
+
         internal void Flush()
         {
             // to reduce the number of updateGauge stats, we only emit queue length on flush
             _metrics.ReporterQueueLength.Update(_commandQueue.Count);
 
             // We can safely drop FlushCommand when the queue is full - sender should take care of flushing
             // in such case
-            _commandQueue.Post(new FlushCommand(this));
-        }
-
-        private async Task FlushLoop()
-        {
-            // First flush should happen later so we start with the delay
-            do
-            {
-                await Task.Delay(_flushInterval).ConfigureAwait(false);
-                Flush();
-            }
-            while (!_commandQueue.Completion.IsCompleted);
+            var flushCommand = new FlushCommand(this);
```

Comment:
This must only add the command to the processing queue as only the processor should flush.

Reply:
If the server is down and we have reached the maximum number of queues. How is our update command executed?

Reply:
It will just skip the flush and add it in the next `FlushLoop` cycle. Since the `FlushCommand` is blocking the `ProcessQueueLoop` until it returns, once the `Sender.Flush` is timing out properly, that `FlushCommand` will return properly and the queue is emptied with all remaining `AppendCommand` and `FlushCommand` being executed right afterwards.